pennylane/optimize/qnspsa.py

```diff
@@ -433,7 +433,10 @@ def _apply_blocking(self, cost, args, kwargs, params_next):
         tape_loss_next = cost.tape.copy(copy_operations=True)
 
         loss_curr, loss_next = qml.execute([tape_loss_curr, tape_loss_next], cost.device, None)
-        loss_curr, loss_next = qml.math.squeeze(loss_curr), qml.math.squeeze(loss_next)
+
+        if not qml.active_return():
+            loss_curr, loss_next = qml.math.squeeze(loss_curr), qml.math.squeeze(loss_next)
```

Comment:
Could it be that any items in `loss_curr` or `loss_next` would require squeezing with the new return types?

Reply:
No, because with the new return types the output of a tape with a single `expval` is a scalar.